File: frontend/src/microservices/traveler/objects/TravelerProfileInput.ts
// 本文件定义 traveler 模块的 `TravelerProfileInput`，作为输入模型并提供 JSON 编解码。

import type {
  TravelerBasicInfo,
  TravelerContactInfo,
  TravelerDocumentInfo,
  TravelerPreferenceInfo,
  TravelerSpecialRequirementInfo,
} from './TravelerProfileDetails'

export type TravelerProfileInput = {
  fullName: string
  documentType: string
  documentNumber: string
  phone: string
  birthDate: string
  seatPreference: string
  mealPreference: string
  accessibilityRequestNotes: string | null
  emergencyContactName: string | null
  emergencyContactPhoneNumber: string | null
  isDefaultTraveler: boolean
  basicInfo: TravelerBasicInfo | null
  documentInfo: TravelerDocumentInfo | null
  contactInfo: TravelerContactInfo | null
  preferenceInfo: TravelerPreferenceInfo | null
  specialRequirementInfo: TravelerSpecialRequirementInfo | null
}

export const travelerProfileInputFromJson = (json: string): TravelerProfileInput =>
  JSON.parse(json) as TravelerProfileInput

export const travelerProfileInputToJson = (value: TravelerProfileInput): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/traveler/objects/TravelerProfileInput.scala
// 本文件定义 traveler 模块的输入对象 `TravelerProfileInput`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerProfileInput(
    fullName: String,
    documentType: String,
    documentNumber: String,
    phone: String,
    birthDate: String,
    seatPreference: String,
    mealPreference: String,
    accessibilityRequestNotes: Option[String],
    emergencyContactName: Option[String],
    emergencyContactPhoneNumber: Option[String],
    isDefaultTraveler: Boolean,
    basicInfo: Option[TravelerBasicInfo] = None,
    documentInfo: Option[TravelerDocumentInfo] = None,
    contactInfo: Option[TravelerContactInfo] = None,
    preferenceInfo: Option[TravelerPreferenceInfo] = None,
    specialRequirementInfo: Option[TravelerSpecialRequirementInfo] = None
)

object TravelerProfileInput:
  given sourceEncoder: Encoder[TravelerProfileInput] = deriveEncoder
  given sourceDecoder: Decoder[TravelerProfileInput] = deriveDecoder
